lib/smack/SmackRep.cpp

```diff
@@ -555,24 +560,50 @@ const Expr* SmackRep::lit(const llvm::Value* v) {
     return neg ? Expr::fn(op.str(), integerLit(0UL,width), e) : e;
 
   } else if (const ConstantFP* CFP = dyn_cast<const ConstantFP>(v)) {
-    const APFloat APF = CFP->getValueAPF();
-    std::string str;
-    raw_string_ostream ss(str);
-    ss << *CFP;
-    std::istringstream iss(str);
-    std::string float_type;
-    long integerPart, fractionalPart, exponentPart;
-    char point, sign, exponent;
-    iss >> float_type;
-    iss >> integerPart;
-    iss >> point;
-    iss >> fractionalPart;
-    iss >> sign;
-    iss >> exponent;
-    iss >> exponentPart;
-
-    return Expr::fn("$fp", Expr::lit(integerPart), Expr::lit(fractionalPart),
-      Expr::lit(exponentPart));
+	if (SmackOptions::BitPrecise) {
+		const APFloat APF = CFP->getValueAPF();
```

Comment:
You are using tabs for indentation here. Change that to two spaces instead. You should never use tabs.